ts/main.ts

```diff
@@ -43,5 +43,109 @@ $entryForm.addEventListener('submit', (event: Event): void => {
   data.entries.unshift($formData);
   if (!$image) throw new Error('the img query failed');
   $image.src = 'images/placeholder-image-square.jpg';
-  $entryForm.reset();
+  $form.reset();
+  $ulElement.prepend(renderEntry(data.entries[0]));
+  viewSwap('entries');
+  toggleNoEntries();
+});
+
+// generate and return a DOM tree for the li element
+
+function renderEntry(entry: Entry): HTMLElement {
+  const $liElement = document.createElement('li');
+
+  const $rowDiv = document.createElement('div');
+  $rowDiv.setAttribute('class', 'row');
+  $liElement.appendChild($rowDiv);
+
+  const $columnHalfDiv = document.createElement('div');
+  $columnHalfDiv.setAttribute('class', 'column-half');
+  $rowDiv.appendChild($columnHalfDiv);
+
+  const $columnHalfDiv2 = document.createElement('div');
+  $columnHalfDiv2.setAttribute('class', 'column-half');
+  $rowDiv.appendChild($columnHalfDiv2);
+
+  const $imgElement = document.createElement('img');
+  $imgElement.setAttribute('src', entry.photoUrl);
+  $imgElement.setAttribute('alt', 'journal entry subject');
+  $columnHalfDiv.appendChild($imgElement);
+
+  const $h2Element = document.createElement('h2');
+  $columnHalfDiv2.appendChild($h2Element);
+  const titleValue = entry.title;
+  $h2Element.textContent = titleValue;
+
+  const $pElement = document.createElement('p');
+  $columnHalfDiv2.appendChild($pElement);
+  const notesValue = entry.notes;
+  $pElement.textContent = notesValue;
+
+  console.log('titleValue: ', titleValue);
+
+  return $liElement;
+}
+
+// add event listener which listens for DOMContentLoaded event
+
+const $ulElement = document.querySelector('ul') as HTMLElement;
+if (!$ulElement) throw new Error(`the 'ul' query failed`);
+
+document.addEventListener('DOMContentLoaded', () => {
+  for (let i = 0; i < data.entries.length; i++) {
+    const newLi = renderEntry(data.entries[i]);
+    $ulElement.appendChild(newLi);
+    viewSwap(data.view);
+    toggleNoEntries();
```

Comment:
These should be outside the `for` loop but still inside the function, so they don't get called on every iteration